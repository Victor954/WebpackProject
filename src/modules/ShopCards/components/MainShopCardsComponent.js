import React from 'react';
import Template from './MainShopCardsView.jsx';

export class MainShopCardsComponent extends React.Component {

    constructor(props){
        super(props);
        this.state = {
            postData: null
        }
    }

    componentDidMount() {

        this.handleLoadData(
            require('../models/PostCollection.json')
        );
    }

    handleLoadData (postData) {

        this.setState({
            postData: postData
        });
    }

    render() {
        return Template.call(this);
    }
  }
