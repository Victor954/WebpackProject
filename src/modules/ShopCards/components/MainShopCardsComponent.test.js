import React from 'react';
import { MainShopCardsComponent } from './MainShopCardsComponent';

import {configure , shallow , mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

let component;
const setUp = (props) => shallow(<MainShopCardsComponent { ...props }/>);

beforeEach(() => {

    component = setUp();
});

afterEach(() => {

    component.unmount();
})

describe('MainShopCardsComponent main functional', () => {

    test('component is rendering' , () => {
        
        expect(component).toMatchSnapshot();
    });

    test('component is not contains void', () => {

        expect(component.children().length).not.toBe(0);
    });
});

describe('MainShopCardsComponent instance functional' , () => {

    let instance;
    const componentDidMountSpy = jest.spyOn(MainShopCardsComponent.prototype, 'componentDidMount')

    beforeEach(() => {

        instance = component.instance();
    });
    
    afterEach(() => {
        componentDidMountSpy.mockReset();
    })

    test('handleLoadData is changing state.postData' , () => {
        const data = [1, 2, 3, 4, 5];

        instance.handleLoadData(data);
        expect(component.state().postData).toEqual(data);
    });

    test('componentDidMountSpy called once time' , () => {
        expect(componentDidMountSpy).toBeCalledTimes(1);
    });

    test('when call componentDidMountSpy called handleLoadData once time' , () => {

        const handleLoadDataSpy = jest.spyOn(instance, 'handleLoadData');
        instance.handleLoadData([]);

        component.update();

        expect(handleLoadDataSpy).toHaveBeenCalledTimes(1);
    });
});