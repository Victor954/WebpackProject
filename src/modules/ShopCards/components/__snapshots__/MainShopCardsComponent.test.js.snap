// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainShopCardsComponent main functional component is rendering 1`] = `ShallowWrapper {}`;
